healme.lic

```diff
@@ -80,24 +88,42 @@ class HealMe
   def cure_disease(spells, health)
     return unless health['diseased']
     return unless spells['Cure Disease']
-
     Flags.add('hm-disease', 'You feel completely cured of all disease', 'you don.t seem to be so afflicted')
-    DRCA.cast_spell(spells['Cure Disease'], @settings, true) until Flags['hm-disease']
-    Flags.delete('hm-disease')
+    loop do 
+	    if Flags['hm-disease']
+	      Flags.delete('hm-disease')
+	      return
+	    else
+	      if !DRSpells.active_spells['Cure Disease']
+          DRCA.cast_spell(spells['Cure Disease'], @settings, true)
+		  end
+	  end
+	end
+  
+    
   end
 
   def flush_poisons(spells, health)
     return unless health['poisoned']
     return unless spells['Flush Poisons']
-
     Flags.add('hm-poison', '^A sudden wave of heat washes over you as your spell attempts to flush poison from your body but finds no toxins', '^A sudden wave of heat washes over you as your spell flushes all poison from your body')
-    DRCA.cast_spell(spells['Flush Poisons'], @settings, true) until Flags['hm-poison']
-    Flags.delete('hm-poison')
+    loop do
+	    if Flags['hm-poison']
+	      Flags.delete('hm-poison')
+	      return
+	    else
+	      if !DRSpells.active_spells['Flush Poisons']
+	        DRCA.cast_spell(spells['Flush Poisons'], @settings, true)
+		    end
+      end
+	  end  
   end
 
   def heal_self(skip_parts)
     return unless DRStats.empath?
-
+    health = check_health
+    cure_disease(spells, health)
```

Comment:
❓ Are diseases and poisons more urgent to heal than wounds?

Reply:
depends on level, if you are taking damage you may never get out of healing until the disease or poison where off

Reply:
played with it with critical poison and it kept in a healing loop because the poison was doing damage